File: services/agent-engine/app/schemas/afm.py
"""Schemas for AFM — Agent File Management."""

from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class AfmAgent(BaseModel):
    agent_id: str
    name: str
    emoji: str = ""
    workspace: str = ""
    target: str = ""
    managed: bool = False


class AfmTeam(BaseModel):
    team_id: Optional[int] = None
    name: str
    folder: str
    exists: bool = False
    members: list[str] = []


class AfmStatusResponse(BaseModel):
    ok: bool
    message: str = ""
    root: str
    is_default: bool
    configured: bool
    agents: list[AfmAgent] = []
    teams: list[AfmTeam] = []


class AfmApplyRequest(BaseModel):
    root: Optional[str] = None  # None/empty → the Evano workspace default


class AfmApplyResult(BaseModel):
    ok: bool
    message: str = ""
    moved: list[str] = []
    skipped: list[str] = []


class AfmArchiveStep(BaseModel):
    agent_id: str = ""
    output: str = ""


class AfmArchiveRequest(BaseModel):
    team_name: str
    steps: list[AfmArchiveStep] = []
    final: str = ""


class AfmArchiveResult(BaseModel):
    ok: bool
    folder: str = ""
    message: str = ""

File: apps/desktop/src/renderer/src/lib/api/types/afm.ts
/**
 * AFM — Agent File Management. Mirrors `services/agent-engine/app/schemas/afm.py`.
 */

export interface AfmAgent {
  agent_id: string;
  name: string;
  emoji: string;
  workspace: string;
  target: string;
  managed: boolean;
}

export interface AfmTeam {
  team_id: number | null;
  name: string;
  folder: string;
  exists: boolean;
  members: string[];
}

/** Response from `GET /afm/status`. */
export interface AfmStatus {
  ok: boolean;
  message: string;
  root: string;
  is_default: boolean;
  configured: boolean;
  agents: AfmAgent[];
  teams: AfmTeam[];
}

/** Response from `POST /afm/apply`. */
export interface AfmApplyResult {
  ok: boolean;
  message: string;
  moved: string[];
  skipped: string[];
}

/** Response from `POST /afm/archive-team-run`. */
export interface AfmArchiveResult {
  ok: boolean;
  folder: string;
  message: string;
}
